src/App.jsx

```diff
@@ -1,36 +1,195 @@
 import "./App.css";
 import TodoList from "./features/TodoList/TodoList.jsx";
 import TodoForm from "./features/TodoForm.jsx";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
 function App() {
   const [todoList, setTodoList] = useState([]);
-  function handleAddTodo(title) {
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [isSaving, setIsSaving] = useState(false);
+  const url = `https://api.airtable.com/v0/${import.meta.env.VITE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+  const token = `Bearer ${import.meta.env.VITE_PAT}`;
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading({ isLoading: true });
+      const options = { method: "GET", headers: { Authorization: token } };
+      try {
+        const resp = await fetch(url, options);
+        if (!resp.ok) {
+          throw new Error(resp.message);
+        }
+        const { records } = await resp.json();
+        const fetchedTodos = records.map((record) => {
+          const todo = {
+            id: record.id,
+            ...record.fields,
+          };
+          if (!todo.isCompleted) {
+            todo.isCompleted = false;
+          }
+          return todo;
+        });
+        setTodoList([...fetchedTodos]);
+      } catch (error) {
+        setErrorMessage(error.message);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchTodos();
+  }, []);
+
+  async function handleAddTodo(title) {
     const newTodo = { title: title, id: Date.now(), isCompleted: false };
     setTodoList([...todoList, newTodo]);
+    const payload = {
```

Comment:
You are calling `setTodoLIst` twice in this function. Since this is supposed to take an pessimistic approach one needs to be removed.